sdk/ai/azure-ai-inference/azure/ai/inference/_patch.py

```diff
@@ -73,6 +96,159 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+_P = ParamSpec("_P")
+_R = TypeVar("_R")
+
+
+def _set_attribute(span: AbstractSpan, key: str, value: Any) -> None:
+    if value is not None:
```

Comment:
I'd prefer to not silently ignore `None`. What fails where if you do set an attribute value to `None`?

Reply:
Ah, I see. Because of how `get` the value from the request/response....

Reply:
The guard against None here is actually because opentelemetry doesn't support it: 

Invalid type NoneType for attribute 'gen_ai.system' value. Expected one of ['bool', 'str', 'bytes', 'int', 'float'] or a sequence of those types

Reply:
I would either rename the method to something like `_set_ifnotnone` (name need massaging) to indicate that part of the purpose in life for it is to ignore `None` or let the same restriction on type flow to the parameter…